service/graph_test.go

```diff
@@ -1217,7 +1216,7 @@ func TestGraphBuildErrors(t *testing.T) {
 					Exporters:  []component.ID{component.NewIDWithName("nop", "conn")},
 				},
 			},
-			expectedStartsWith: "topo: no topological ordering: cyclic components:",
+			expected: `cycle detected: connector "nop/conn" -> processor "nop" -> connector "nop/conn"`,
```

Comment:
Print incoming pipeline type and outgoing for conn?

Reply:
I added this info. Also, since processors are associated with a specific pipeline, I added that ID as well.

Let me know if you have suggestions for the format.